### PR TITLE
Add docs for singleVariant question option

### PR DESCRIPTION
**Pull request by mwest1066**

### CODE REVIEW

File: doc/question.md [hunk 1]
```diff
@@ -97,14 +101,29 @@ The `question.html` is a template used to render the question to the student. A
...
-The `question.html` is regular HTML, with three special features:
+The `question.html` is regular HTML, with four special features:
 
 1. Any text in double-curly-braces (like `{{params.m}}`) is substituted with variable values. This is using [Mustache](https://mustache.github.io/mustache.5.html) templating.
 
 2. Special HTML elements (like `<pl-number-input>`) enable input and formatted output. See the [list of PrairieLearn elements](elements.md).
    
 3. A special `<markdown>` tag allows you to write Markdown inline in questions.
 
+4. LaTeX equations are available by putting `$y = x^2$` or `\[ y = \int_0^1 x^2\ dx \]` into the HTML.
```
Comment (coatless) [inline doc/question.md]: How about a distinction between `inline` and `display`?

```
Inline: `$y = x^2$` or `\(a^2+ b^2 = c^2\)`
Display: 
`$$\phi = \frac{1 + \sqrt{5}}{2}$$`
`\[ y = \int_0^1 x^2\ dx \]`

File: doc/question.md [hunk 2]
```diff
@@ -97,14 +101,29 @@ The `question.html` is a template used to render the question to the student. A
...
+
+## The `singleVariant` option for non-randomized questions
+
+While it is recommended that all questions contain random parameters, sometimes it is impractical to do this. For questions that don't have a meaningful amount of randomization in them, the `info.json` file should set `"singleVariant": true`. This has the following effects:
+
+* On Homeworks, each student will only ever be given one variant of the question, which they can repeatedly attempt without limit. The correct answer will never be shown to students.
```
Comment (coatless) [inline doc/question.md]: Bold homework and exam? Emphasize this is a `type`?

File: doc/question.md [hunk 3]
```diff
@@ -97,14 +101,29 @@ The `question.html` is a template used to render the question to the student. A
...
+* On Exams, the `singleVariant` option has no effect and the question is treated like any other.
+
+## The `partialCredit` option
+
+By default all questions award partial credit. For example, if there are two numeric answers in a question and only one of them is correct then the student will be awarded 50% of the available points.
+
+To disable partial credit for a question, set `"partialCredit": false` in the `info.json` file for the question. This will mean that the question will either give 0% or 100%, and it will only give 100% if every element on the page is fully correct.
```
Comment (coatless) [inline doc/question.md]: Maybe mention that some elements (e.g. `pl-checkbox`) have more fine grain control over partial credit being awarded?

File: doc/question.md [hunk 4]
```diff
@@ -45,22 +47,24 @@ The `info.json` file for each question defines properties of the question. For e
...
-
-## Partial credit
-
-By default all v3 questions award partial credit. For example, if there are two numeric answers in a question and only one of them is correct then the student will be awarded 50% of the available points.
+* [Format specification for question `info.json`](https://github.com/PrairieLearn/PrairieLearn/blob/master/schemas/schemas/infoQuestion.json)
 
-To disable partial credit for a question, set `"partialCredit": false` in the `info.json` file for the question. This will mean that the question will either give 0% or 100%, and it will only give 100% if every element on the page is fully correct.
+Property | Type | Description
+--- | --- | ---
+`uuid` | string | Unique identifier (UUID v4). E.g., `"8b4891d6-64d1-4e89-b72d-ad2133f25b2f"`. These can be obtained from https://www.uuidgenerator.net (Required: no default)
```
Comment (nwalters512) [inline doc/question.md]: Nit: would prefer semicolons here; for cases like `Optional; default: none`, I think that's a little clearer than `Optional: default: none`.

```suggestion
`uuid` | string | Unique identifier (UUID v4). E.g., `"8b4891d6-64d1-4e89-b72d-ad2133f25b2f"`. These can be obtained from https://www.uuidgenerator.net (Required; no default)
```

File: doc/question.md (hunk 2 above)
Comment (nwalters512) [reply doc/question.md]: Maybe phrase it as "Homework-type assessments"?

File: doc/question.md [hunk 5]
```diff
@@ -97,14 +101,29 @@ The `question.html` is a template used to render the question to the student. A
...
+## The `singleVariant` option for non-randomized questions
+
+While it is recommended that all questions contain random parameters, sometimes it is impractical to do this. For questions that don't have a meaningful amount of randomization in them, the `info.json` file should set `"singleVariant": true`. This has the following effects:
+
+* On Homeworks, each student will only ever be given one variant of the question, which they can repeatedly attempt without limit. The correct answer will never be shown to students.
+* On Exams, the `singleVariant` option has no effect and the question is treated like any other.
+
+## The `partialCredit` option
+
+By default all questions award partial credit. For example, if there are two numeric answers in a question and only one of them is correct then the student will be awarded 50% of the available points.
```
Comment (nwalters512) [inline doc/question.md]:
```suggestion
By default, all questions award partial credit. For example, if there are two numeric answers in a question and only one of them is correct, the student will be awarded 50% of the available points.
```

File: doc/question.md [hunk 6]
```diff
@@ -45,22 +47,24 @@ The `info.json` file for each question defines properties of the question. For e
...
-- `title` gives a student-visible title for the question.
-- `topic` is the part of the course that this question belongs to (like the chapter in a textbook).
-- `tags` (optional) stores any other aspects of the questions, for sorting and searching (these can be anything).
-- `clientFiles` (optional) lists the files that the client (student's webbrowser) can access. Anything in here should be considered viewable by the student.
-- `type` specifies the question format and should be `"v3"` for the current PrairieLearn question format.
-
-## Partial credit
-
-By default all v3 questions award partial credit. For example, if there are two numeric answers in a question and only one of them is correct then the student will be awarded 50% of the available points.
+* [Format specification for question `info.json`](https://github.com/PrairieLearn/PrairieLearn/blob/master/schemas/schemas/infoQuestion.json)
```
[the remark this answers is not in the document]
Comment (mwest1066) [reply doc/question.md]: done

File: doc/question.md [hunk 7]
```diff
@@ -97,14 +101,29 @@ The `question.html` is a template used to render the question to the student. A
 <p>$F = $ <pl-number-input answers_name="F" comparison="sigfig" digits="2" /> $\rm m/s^2$
 ```
 
-The `question.html` is regular HTML, with three special features:
+The `question.html` is regular HTML, with four special features:
 
 1. Any text in double-curly-braces (like `{{params.m}}`) is substituted with variable values. This is using [Mustache](https://mustache.github.io/mustache.5.html) templating.
```
[the remark this answers is not in the document]
Comment (mwest1066) [reply doc/question.md]: done

File: doc/question.md (hunk 1 above)
Comment (mwest1066) [reply doc/question.md]: done

File: doc/question.md (hunk 2 above)
Comment (mwest1066) [reply doc/question.md]: good idea, done

File: doc/question.md (hunk 3 above)
Comment (mwest1066) [reply doc/question.md]: done

File: doc/question.md (hunk 4 above)
Comment (mwest1066) [reply doc/question.md]: good catch, this was just an error 😄

File: doc/question.md (hunk 5 above)
Comment (mwest1066) [reply doc/question.md]: done

File: doc/question.md [hunk 8]
```diff
@@ -45,22 +47,24 @@ The `info.json` file for each question defines properties of the question. For e
     "uuid": "cef0cbf3-6458-4f13-a418-ee4d7e7505dd",
     "title": "Addition of vectors in Cartesian coordinates",
     "topic": "Vectors",
-    "tags": ["Cartesian", "graphical"],
+    "tags": ["secret", "graphical"],
```
[the remark this answers is not in the document]
Comment (mwest1066) [reply doc/question.md]: I changed it to be consistent with the example `question.html` and `server.py`, although this is not currently a question in the example course.

**Pull request merged**